cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -507,7 +524,23 @@ teeForge ft tverb tr = Tracer $ \ev -> do
       Consensus.TraceBlockFromFuture{} -> teeForge' (ftTraceBlockFromFuture ft)
       Consensus.TraceSlotIsImmutable{} -> teeForge' (ftTraceSlotIsImmutable ft)
       Consensus.TraceNodeIsLeader{} -> teeForge' (ftTraceNodeIsLeader ft)
-  traceWith (toLogObject' tverb tr) ev
+  case event of
+    Consensus.TraceStartLeadershipCheck slot -> do
+      utxoSize <- mapNodeKernelDataIO nkUtxoSize nodeKern
```

Comment:
Let's add a bang here (or in `nkUtxoSize`) to force it as soon as possible, because you're holding on to the whole ledger state, which is huge.